java/src/test/java/ai/onnxruntime/SparseTensorTest.java

```diff
@@ -0,0 +1,450 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * Licensed under the MIT License.
+ */
+package ai.onnxruntime;
+
+import static ai.onnxruntime.InferenceTest.getResourcePath;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.nio.Buffer;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.FloatBuffer;
+import java.nio.LongBuffer;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+
+public class SparseTensorTest {
+
+  @Test
+  public void testCSRC() throws OrtException {
+    String modelPath = getResourcePath("/generic_sparse_to_dense_matmul.onnx").toString();
```

Comment:
>"/generic_sparse_to_dense_matmul.onnx [](http://example.com/codeflow?start=39&length=37)

There is an existing model that is used in the unit tests for the same purpose. It would be good to verify Java interface works the same way. Also we usually check in the script that generates the model.

Reply:
I added `generic_sparse_to_dense_matmul.onnx` to remove the size restriction on the inputs and outputs in the existing `sparse_to_dense_matmul.onnx` which allowed me to test multiple different indices structures without making more models. I honestly don't remember how I created this model, it might have been at the jshell prompt writing the protobuf directly. I can make a static method on `SparseTensorTest` that creates it if you want.

The one that produces a sparse tensor as an output is `sparse_initializer_as_output.onnx` and I use that later in this test file. If there are more models which I should be using to test this code let me know, I couldn't spot any others.